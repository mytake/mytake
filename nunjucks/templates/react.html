{% extends "base.html" %}

{% block endbody %}
  <!-- Dependencies -->
  <script src="https://unpkg.com/react@15/dist/react.js"></script>
  <script src="https://unpkg.com/react-dom@15/dist/react-dom.js"></script>

  <!-- Main -->
  <script src="./bundle-0.0.3.js"></script>
{% endblock %}
