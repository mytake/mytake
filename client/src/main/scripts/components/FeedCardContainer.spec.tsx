import * as React from "react";
import * as renderer from "react-test-renderer";
import {
  FeedCardBranch,
  FeedCardContainerProps,
  FeedCardContainerState
} from "./FeedCardContainer";
import {
  documentFactLink,
  documentNodes,
  videoFactFast
} from "../utils/testUtils";

const videoProps: FeedCardContainerProps = {
  username: "samples",
  titleSlug: "why-its-so-hard-to-have-peace",
  title: "Why it's so hard to have peace",
  blocks: [
    {
      kind: "paragraph",
      text:
        "Today, anybody who wants to negotiate with a Muslim country is “weak”."
    },
    {
      kind: "video",
      range: [304, 321],
      videoId: "iEfwIxM7MmnOKb7zt4HqW8IxUWy6F7a236fSOQlUUWI="
    }
  ]
};

const documentProps: FeedCardContainerProps = {
  username: "samples",
  titleSlug: "does-a-law-mean-what-it-says-or-what-it-meant",
  title: "Does a law mean what it says, or what it meant?",
  blocks: [
    {
      kind: "document",
      excerptId: "pMHhbW_I-wquOfoyPFAVQu8DMLMpYVxhGT8R1x71hYA=",
      viewRange: [283, 439],
      highlightedRange: [283, 439]
    },
    {
      kind: "paragraph",
      text:
        "If our judges get to decide that the law means whatever they want it to mean, then we still live under a King, we’ve just changed his name to Judge. That’s why it’s important to apply consistent methodology when interpreting the law. But in this case, whether you’re a Textualist or an Originalist, I don’t see how the 2nd amendment allows for any of the gun restrictions in place today."
    }
  ]
};

test("Card loading", () => {
  const containerProps = documentProps;
  const containerState: FeedCardContainerState = {
    loading: true
  };

  const tree = renderer
    .create(
      <FeedCardBranch
        containerProps={containerProps}
        containerState={containerState}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Card w/ Document Fact", () => {
  const containerProps = documentProps;
  const containerState: FeedCardContainerState = {
    loading: false,
    document: {
      fact: documentFactLink.fact,
      nodes: documentNodes
    }
  };

  const tree = renderer
    .create(
      <FeedCardBranch
        containerProps={containerProps}
        containerState={containerState}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Card w/ Video Fact", () => {
  const containerProps = videoProps;
  const containerState: FeedCardContainerState = {
    loading: false,
    videoFact: videoFactFast
  };

  const tree = renderer
    .create(
      <FeedCardBranch
        containerProps={containerProps}
        containerState={containerState}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
