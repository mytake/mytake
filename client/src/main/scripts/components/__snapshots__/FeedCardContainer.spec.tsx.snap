// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card loading 1`] = `
<div
  className="feed__card"
>
  <h2
    className="feed__card-title"
  >
    Loading Take Preview
  </h2>
</div>
`;

exports[`Card w/ Document Fact 1`] = `
<div
  className="feed__card"
>
  <a
    className="feed__link"
    href="/samples/does-a-law-mean-what-it-says-or-what-it-meant"
  >
    <span />
  </a>
  <div
    className="feed__card-thumb-container feed__card-thumb-container--document"
  >
    <div
      className="feed__card-document"
    >
      <h2
        className="feed__card-document-title"
      >
        Amendment 13
      </h2>
      <div
        className="feed__card-document-text"
      >
        <p
          data-char-offset={218}
        >
          Section 2. Congress shall have power to enforce this article by appropriate legislation.
        </p>
      </div>
    </div>
  </div>
  <div
    className="feed__card-excerpt"
  >
    <h2
      className="feed__card-title"
    >
      Does a law mean what it says, or what it meant?
    </h2>
    <div
      className="feed__card-text-container"
    >
      <p
        className="feed__card-text"
      >
        If our judges get to decide that the law means whatever they want it to mean, then we still live under a King, we’ve just changed his name to Judge. That’s why it’s important to apply consistent methodology when interpreting the law. But in this case, whether you’re a Textualist or an Originalist, I don’t see how the 2nd amendment allows for any of the gun restrictions in place today.
      </p>
    </div>
  </div>
  <span
    className="feed__more"
  >
    Read more
  </span>
</div>
`;

exports[`Card w/ Video Fact 1`] = `
<div
  className="feed__card"
>
  <a
    className="feed__link"
    href="/samples/why-its-so-hard-to-have-peace"
  >
    <span />
  </a>
  <div
    className="feed__card-thumb-container feed__card-thumb-container--video"
  >
    <img
      className="feed__card-thumb"
      src="https://img.youtube.com/vi/ApTLB76Nmdg/0.jpg"
    />
  </div>
  <div
    className="feed__card-excerpt"
  >
    <h2
      className="feed__card-title"
    >
      Why it's so hard to have peace
    </h2>
    <div
      className="feed__card-text-container"
    >
      <p
        className="feed__card-text"
      >
        Today, anybody who wants to negotiate with a Muslim country is “weak”.
      </p>
    </div>
  </div>
  <span
    className="feed__more"
  >
    Read more
  </span>
</div>
`;
